payments-ui-core/src/main/java/com/stripe/android/ui/core/elements/PostalCodeConfig.kt

```diff
@@ -0,0 +1,80 @@
+package com.stripe.android.ui.core.elements
+
+import androidx.annotation.StringRes
+import androidx.compose.ui.text.input.KeyboardCapitalization
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.VisualTransformation
+import kotlinx.coroutines.flow.MutableStateFlow
+
+internal class PostalCodeConfig(
+    @StringRes override val label: Int,
+    override val capitalization: KeyboardCapitalization = KeyboardCapitalization.Words,
+    override val keyboard: KeyboardType = KeyboardType.Text,
+    override val trailingIcon: MutableStateFlow<TextFieldIcon?> = MutableStateFlow(null),
+    country: String
+) : TextFieldConfig {
+    override val debugLabel: String = "postal_code_text"
+    override val visualTransformation: VisualTransformation? = null
+    override val loading: MutableStateFlow<Boolean> = MutableStateFlow(false)
+    private val format = CountryPostalFormat.forCountry(country)
+
+    override fun determineState(input: String): TextFieldState = object : TextFieldState {
+        override fun shouldShowError(hasFocus: Boolean) = false
+
+        override fun isValid(): Boolean {
+            return input.length in format.minimumLength..format.maximumLength &&
+                input.matches(format.regexPattern)
+        }
+
+        override fun getError(): FieldError? = null
+
+        override fun isFull(): Boolean = input.length >= format.minimumLength
+
+        override fun isBlank(): Boolean = input.isBlank()
+    }
+
+    override fun filter(userTyped: String): String =
+        if (
+            setOf(KeyboardType.Number, KeyboardType.NumberPassword).contains(keyboard)
+        ) {
+            userTyped.filter { it.isDigit() }
+        } else {
+            userTyped
+        }.replace(Regex("\\s*"), "")
```

Comment:
On iOS, it looks like we restrict the input length to the country’s maximum length. Can we do the same here?